dask/dataframe/tests/test_dataframe.py

```diff
@@ -5109,3 +5109,38 @@ def reduce_fn(series):
         .compute()[0]
     )
     assert result == {"x": 14, "y": 64}
+
+
+@pytest.fixture
+def demo_ts():
+    return dd.demo.make_timeseries()
+
+
+def test_set_index_with_dask_dt_index_gh8359(demo_ts):
```

Comment:
As a general note, I don't think we need to name tests after GitHub issues unless they are testing for extremely weird behavior. In this case, it's a fairly standard usage of the API, so just describing the test with the name is probably enough.